toolchain/semantics/semantics_parse_tree_handler.cpp

```diff
@@ -58,7 +59,9 @@ auto SemanticsParseTreeHandler::Build() -> void {
   node_block_stack_.push_back(semantics_->AddNodeBlock());
   PushScope();
 
-  for (auto parse_node : parse_tree_->postorder()) {
+  while (postorder_it_ != postorder_end_) {
+    auto parse_node = *postorder_it_;
+    ++postorder_it_;
```

Comment:
Is there a reason to write this as a `while` rather than a `for` loop? Eg:
```suggestion
  for (/*iterator initialized in constructor*/; postorder_it_ != postorder_end_; ++postorder_it_) {
    auto parse_node = *postorder_it_;
```

Reply:
Reverted this part of the change as part of removing NextParseNodeIs